sdk/python/kfp/dsl/_pipeline_param.py

```diff
@@ -48,7 +48,13 @@ def _extract_pipelineparams(payloads: str or list[str]):
   matches = []
   for payload in payloads:
     matches += _match_serialized_pipelineparam(payload)
-  return [PipelineParam(x[1], x[0], x[2]) for x in list(set(matches))]
+  pipeline_params = []
+  for x in list(set(matches)):
```

Comment:
nit: I really think it's not a good coding practice to return a matches from _match_serialized_pipelineparam. It should return a custom type which converts match to strong typed data. It's very hard to maintain the magic number index in the future. You don't need to fix this comment in the PR.

Reply:
make sense. Done in the #975